docs/source/user_guide.rst

```diff
@@ -0,0 +1,606 @@
+    print(response.answer)
+
+
+Using a custom model
+-----------------------------
+
+If you do not wish to use LangChain, you can create your own LLM class by subclassing
+the LLMInterface. Here is an example using the Python ollama client:
+
+.. code:: python
+
+    import ollama
+    from neo4j_genai.llm import LLMInterface, LLMResponse
+
+    class OllamaLLM(LLMInterface):
+
+        def invoke(self, input: str) -> LLMResponse:
+            response = ollama.chat(model=self.model_name, messages=[
+              {
+                'role': 'user',
+                'content': input,
+              },
+            ])
+            return LLMResponse(
+                content=response["message"]["content"]
+            )
+
+    # retriever = ...
+
+    llm = OllamaLLM("llama3:8b")
+
+    rag = GraphRAG(retriever=retriever, llm=llm)
+    query_text = "How do I do similarity search in Neo4j?"
+    response = rag.search(query_text=query_text, retriever_config={"top_k": 5})
+    print(response.answer)
+
+Also see :ref:`llminterface`.
+
+
+Configuring the prompt
+========================
+
+Prompt are managed through `PromptTemplate` classes. More
+specifically, the `GraphRAG` pipeline uses a `RagTemplate` with
+a default prompt. You can use another prompt by subclassing
+the `RagTemplate` class and passing it to the `GraphRAG` pipeline
+object during initialization:
+
+.. code:: python
+
+    from neo4j_genai.generation import RagTemplate, GraphRAG
+
+    # retriever = ...
+    # llm = ...
+
+    prompt_template = RagTemplate(
+        prompt="Answer the question {question} using context {context} and examples {examples}",
+        expected_inputs=["context", "question", "examples"]
+    )
+
+    rag = GraphRAG(retriever=retriever, llm=llm, prompt_template=prompt_template)
+
+    # ...
+
+
+Also see :ref:`prompttemplate`.
+
+
+The last piece you can configure in the GraphRAG pipeline is the retriever. The different options
+are described below.
+
+
+************************
+Retriever Configuration
+************************
+
+We provide implementation for the most commonly used retrievers:
+
+1. Vector retriever: performs a similarity search based on a Neo4j vector index and a query text or vector. Returns the matched node.
+2. Vector Cypher retriever: performs a similarity search based on a Neo4j vector index and a query text or vector. Returned results can be configured through a retrieval query parameter that will be executed after the index search. It can be used to fetch more context around the matched node.
+3. External retrievers: use these retrievers when your vectors are not saved in Neo4j but in an external vector database. We currently support Weaviate and Pinecone vector databases.
+4. Hybrid and Hybrid cypher retrievers: these retrievers use both a vector and full-text indexes.
+
+Retrievers all expose a `search` method that we will discuss in the next sections.
```

Comment:
Should we comment saying that this is an approximate nearest neighbor search and link to https://neo4j.com/docs/cypher-manual/current/indexes/semantic-indexes/vector-indexes/#_limitiations_and_known_issues ? We also have this in the index page and README

Reply:
Basically so that developers are aware that this isn't an exact search